zerver/lib/actions.py

```diff
@@ -2737,6 +2763,28 @@ def our_person(p):
                         bot.update(event['bot'])
 
         elif event['type'] == 'stream':
+            if event['op'] == 'create':
+                for stream in event['streams']:
+                    stream_name = stream['name']
+                    realm = get_realm('zulip.com')
+                    new_stream, created = create_stream_if_needed(realm, stream_name, stream['invite_only'])
```

Comment:
This might be possible that the event is to create a new stream with name of already existing stream. So, if streams with same name already exist this won't create a new stream.